src/components/Post/RandomButton.tsx

```diff
@@ -0,0 +1,36 @@
+import { useRouter } from 'next/router';
+
+import { randomSlug } from 'helpers/randomSlug';
+
+export const postRandomButtonTestIds = {
+  root: 'posts__random-button',
+};
+
+const RandomButton = () => {
+  const router = useRouter();
+  const onClick = async () => {
+    const slug = await randomSlug();
+
+    if (slug === '') {
+      router.push('/');
+
+      return;
+    }
+
+    router.push(`/posts/${slug}`);
+  };
+
+  return (
+    <div className="absolute top-0 right-0 p-4">
+      <button
+        className="btn btn-circle"
+        onClick={onClick}
+        data-test-id={postRandomButtonTestIds.root}
+      >
+        <span className="random-button__text">TIL</span>
```

Comment:
Should we omit the `span` tag?
Or we will use it for some purpose?

Reply:
I've just updated to hide the text when it's loading :D